### PR TITLE
[AppGw] Introducing new Property - Header Value Matcher

### PR DESCRIPTION
**Pull request by Sukhdev841**
## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] N/A

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Accounts/Accounts/Accounts.csproj [hunk 1]
```diff
@@ -20,6 +20,7 @@
   </ItemGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.Azure.Management.Network" Version="26.0.0" />
```
Comment (Nickcandy) [inline src/Accounts/Accounts/Accounts.csproj:23]: Why add this dependency in accounts?

File: src/Network/Network/ApplicationGateway/RewriteRuleHeaderConfiguration/AzureApplicationGatewayRewriteRuleHeaderConfigurationBase.cs [hunk 2]
```diff
@@ -32,6 +32,11 @@ public class AzureApplicationGatewayRewriteRuleHeaderConfigurationBase : Network
             HelpMessage = "Header value to set for the given header name. Header will be deleted if this is omitted")]
         public string HeaderValue { get; set; }
 
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "[ToDo] Provide help message")]
```
Comment (Nickcandy) [inline src/Network/Network/ApplicationGateway/RewriteRuleHeaderConfiguration/AzureApplicationGatewayRewriteRuleHeaderConfigurationBase.cs:37]: should this message be populated before merge ?

**Comment by YanaXu** [comment]
Hi @Sukhdev841, could you check [Nickcandy](https://github.com/Nickcandy)'s comment and resolve the conflict?

File: src/Network/Network/ApplicationGateway/RewriteRuleHeaderConfiguration/AzureApplicationGatewayRewriteRuleHeaderConfigurationBase.cs (hunk 2 above)
Comment (Sukhdev841) [reply src/Network/Network/ApplicationGateway/RewriteRuleHeaderConfiguration/AzureApplicationGatewayRewriteRuleHeaderConfigurationBase.cs]: change has been made.

**Comment by Sukhdev841** [comment]
Hi @YanaXu / @Nickcandy, I resolved the conflicts and replied to all previous comments. Please have a look.

File: src/Network/Network/help/New-AzApplicationGatewayHeaderValueMatcher.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+```
+
+## DESCRIPTION
+**The New-AzApplicationGatewayHeaderValueMatcher** cmdlet creates a Header Value Matcher object for an Azure application gateway.
+
+## EXAMPLES
+
+### Example 1
+```powershell
+PS C:\> $hvm = New-AzApplicationGatewayHeaderValueMatcher -Pattern ".*" -IgnoreCase -Negate
```
Comment (YanaXu) [inline src/Network/Network/help/New-AzApplicationGatewayHeaderValueMatcher.md:27]:
```suggestion
$hvm = New-AzApplicationGatewayHeaderValueMatcher -Pattern ".*" -IgnoreCase -Negate
```

File: src/Network/Network/help/New-AzApplicationGatewayHeaderValueMatcher.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+PS C:\> $requestHeaderConfiguration01 = New-AzApplicationGatewayRewriteRuleHeaderConfiguration -HeaderName "Set-Cookie" -HeaderValue "val" -HeaderValueMatcher $headerValueMatcher
```
Comment (YanaXu) [inline src/Network/Network/help/New-AzApplicationGatewayHeaderValueMatcher.md:28]:
```suggestion
$requestHeaderConfiguration01 = New-AzApplicationGatewayRewriteRuleHeaderConfiguration -HeaderName "Set-Cookie" -HeaderValue "val" -HeaderValueMatcher $headerValueMatcher
```

**Comment by YanaXu** [comment]
Hi @Sukhdev841, please fix CI errors.

File: src/Network/Network/help/New-AzApplicationGatewayHeaderValueMatcher.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
+---
+external help file: Microsoft.Azure.PowerShell.Cmdlets.Network.dll-Help.xml
+Module Name: Az.Network
+online version:
```
Comment (YanaXu) [inline src/Network/Network/help/New-AzApplicationGatewayHeaderValueMatcher.md:4]: According to the CI errors, you have to set the value of "oneline version".

**Comment by YanaXu** [comment]
Hi @Sukhdev841, please fix the CI tests.

**Review by YanaXu — APPROVED** [review]

**Pull request merged**